doc/api_ref/providers.rst

```diff
@@ -0,0 +1,169 @@
+External Providers
+==============================
+
+Botan ships with a variety of cryptographic algorithms in both pure software
+as well as with support from :doc:`hardware acceleration <../hardware_acceleration>`.
```

Comment:
Should we link to an rST label rather than a file, maybe?

Reply:
If we would want to do that, we would need to add a label "hardware acceleration" to `hardware_acceleration.rst`, which is kind of double the work. After #3849 was merged, we are now able to reference whole documents as well as headlines with autogenerated labels. To see all labels sphinx generates, run `python3 -m sphinx.ext.intersphinx build/docs/handbook/objects.inv`.